src/test/java/com/myhome/services/unit/HouseSDJpaServiceTest.java

```diff
@@ -0,0 +1,235 @@
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+
+class HouseSDJpaServiceTest {
+
+  private final int TEST_HOUSES_COUNT = 10;
+  private final int TEST_HOUSE_MEMBERS_COUNT = 10;
+  private final String HOUSE_ID = "test-house-id";
+  private final String MEMBER_ID = "test-member-id";
+
+  @Mock
+  private HouseMemberRepository houseMemberRepository;
+  @Mock
+  private HouseMemberDocumentRepository houseMemberDocumentRepository;
+  @Mock
+  private CommunityHouseRepository communityHouseRepository;
+  @InjectMocks
+  private HouseSDJpaService houseSDJpaService;
+
+  @BeforeEach
+  void setUp() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  void listAllHousesDefault() {
+    // given
+    Set<CommunityHouse> housesInDatabase = getTestHouses(TEST_HOUSES_COUNT);
+    given(communityHouseRepository.findAll())
+        .willReturn(housesInDatabase);
+
+    // when
+    Set<CommunityHouse> resultHouses = houseSDJpaService.listAllHouses();
+
+    // then
+    assertEquals(housesInDatabase, resultHouses);
+    verify(communityHouseRepository).findAll();
+  }
+
+  @Test
+  void listAllHousesCustomPageable() {
+    // given
+    Set<CommunityHouse> housesInDatabase = getTestHouses(TEST_HOUSES_COUNT);
+    Pageable pageRequest = PageRequest.of(0, TEST_HOUSES_COUNT);
+    Page<CommunityHouse> housesPage = new PageImpl<>(
+        new ArrayList<>(housesInDatabase),
+        pageRequest,
+        TEST_HOUSES_COUNT
+    );
+    given(communityHouseRepository.findAll(pageRequest))
+        .willReturn(housesPage);
+
+    // when
+    Set<CommunityHouse> resultHouses = houseSDJpaService.listAllHouses(pageRequest);
+
+    // then
+    assertEquals(housesInDatabase, resultHouses);
+    verify(communityHouseRepository).findAll(pageRequest);
+  }
+
+  @Test
+  void addHouseMembers() {
+    // given
+    Set<HouseMember> membersToAdd = getTestHouseMembers(TEST_HOUSE_MEMBERS_COUNT);
+    int membersToAddSize = membersToAdd.size();
+    CommunityHouse communityHouse = getTestCommunityHouse();
+
+    given(communityHouseRepository.findByHouseId(HOUSE_ID))
+        .willReturn(communityHouse);
+    given(houseMemberRepository.saveAll(membersToAdd))
+        .willReturn(membersToAdd);
+
+    // when
+    Set<HouseMember> resultMembers = houseSDJpaService.addHouseMembers(HOUSE_ID, membersToAdd);
+
+    // then
+    assertEquals(membersToAddSize, resultMembers.size());
+    assertEquals(membersToAddSize, communityHouse.getHouseMembers().size());
```

Comment:
This assertion is not needed - you are asserting the test data that you created with `getTestCommunityHouse` method